deploy-deps.sh

```diff
@@ -49,7 +49,7 @@ deploy() {
 test_pvc_binding(){
     local pvc_resources="${script_path}/dependencies/pre-deployment-pvc-binding"
     echo "Creating PVC from '$pvc_resources' using the cluster's default storage class"
-    kubectl apply -k "$pvc_resources"
+    retry "kubectl apply -k ${pvc_resources}" "Error while creating pre-deployment-pvc-binding"
```

Comment:
The retries sleep intervals are not enough on my laptop if running the script as soon as the cluster starts.
It might make things better, but it won't work around the issue completely.

The thing is that even if we separate the creation of the namespace and wait on the default service account to be created, I think we'll still need to use retry for that service account creation which will potentially fail in some cases as it fails for me now.

We can maybe introduce a 5 seconds sleep before the retry. I don't like it, but I currently don't see a better alternative.

Reply:
I don't like using sleep either. What about creating a service account other than default with a YAML? I saw you submitted a solution to create the default service account, did you try with another account? If the system that creates the default service account is slow using another service account could solve the problem.

Reply:
Changing the service account did not work. I added a 10 seconds sleep, as 5 seconds was not enough.

Reply:
I'm good with that. Thank you for looking into it!

Reply:
I tried it now, and it failed the first retry out of 3 after waiting the 10 seconds, and then the succeeded on the second. For me it means that 10 seconds is a good choice.